app/services/organizations/delete_service.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module Organizations
+  class DeleteService
+    include Sagittarius::Database::Transactional
+
+    attr_reader :current_user, :organization
+
+    def initialize(current_user, organization)
+      @current_user = current_user
+      @organization = organization
+    end
+
+    def execute
+      unless Ability.allowed?(current_user, :delete_organization, organization)
+        return ServiceResponse.error(message: 'Missing permission', payload: :missing_permission)
+      end
+
+      transactional do
+        organization.delete
+
+        if organization.persisted?
+          return ServiceResponse.error(message: 'Failed to delete organization', payload: organization.errors)
```

Comment:
In this case, we should also roll back the transaction because there could be other things deleted through the cascading foreign keys.
```suggestion
          t.rollback_and_return! ServiceResponse.error(message: 'Failed to delete organization', payload: organization.errors)
```